structure/heap/heap_test.go

```diff
@@ -0,0 +1,100 @@
+package heap
+
+import (
+	"testing"
+)
+
+type _int int
+
+func (u _int) Less(o _int) bool {
+	return u < o
+}
+
+func TestHeap_Empty(t *testing.T) {
+	tests := []struct {
+		name string
+		want bool
+	}{
+		{name: "empty", want: true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := New[_int]()
+			if got := h.Empty(); got != tt.want {
+				t.Errorf("Empty() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+type _opType int
+
+const (
+	_push  = 1
+	_pop   = 2
+	_test  = 3
+	_empty = 4
+)
+
+type _op struct {
+	typ     _opType
+	x       _int
+	isEmpty bool
+}
```

Comment:
Again, please don't use such convention and refer to some community guidelines on naming convention in Go. Example: [Google's style guide](https://google.github.io/styleguide/go/).